daylay92/stacktrace-backend
# Add signup functionality

#### What does this PR do?
Enables users to register on the App.
#### Description of Task to be completed?
- Create model, service, route and controller classes and methods that facilitate the signup functionality.
- Implement utility methods for hashing passwords and generating JWT token, amongst other things.
- Setup openapi documentation with swagger and add signup route descriptions.
- Implement middlewares to ensure only valid data sets are stored in the database.
- Implement validation functions.
- Write unit test.
#### How should this be manually tested?
- Clone the repo and cd into the project folder
- Checkout to `ft-user-signup` branch and run `npm install`.
- Add a `.env` file to the root directory and include in it the following environment variables;
```
 MONGO_DEV_URI=your mongodb connection string for development
MONGO_URI=your mongodb connection string for the test environment
```
- Start the dev server by running `npm run start:dev`
- Using postman or insomnia, make a post request to `localhost:3000/api/v1/auth/signup` with the parameters described below;
```
{
  "firstName": "King",
  "lastName": "David",
  "email": "stacktrace@gmail.com",
  "password": "Elija455h"
}
```
Feel free to change the fields.
- You can also launch the api docs on your browser via `http://localhost:3000/api/v1/docs` for a quick test of the functionality once your dev server is up and runnng.
- You can run `npm test` to see the results of the unit tests.
